clang/lib/Sema/SemaDecl.cpp

```diff
@@ -9203,15 +9203,15 @@ static NamedDecl *DiagnoseInvalidRedeclaration(
         << Idx << FDParam->getType()
         << NewFD->getParamDecl(Idx - 1)->getType();
     } else if (FDisConst != NewFDisConst) {
-      SemaRef.Diag(FD->getLocation(), diag::note_member_def_close_const_match)
-          << NewFDisConst << FD->getSourceRange().getEnd()
-          << (NewFDisConst
-                  ? FixItHint::CreateRemoval(ExtraArgs.D.getFunctionTypeInfo()
-                                                 .getConstQualifierLoc())
-                  : FixItHint::CreateInsertion(ExtraArgs.D.getFunctionTypeInfo()
-                                                   .getRParenLoc()
-                                                   .getLocWithOffset(1),
-                                               " const"));
+      auto DB = SemaRef.Diag(FD->getLocation(),
+                             diag::note_member_def_close_const_match)
+                << NewFDisConst << FD->getSourceRange().getEnd();
+      if (const auto &FTI = ExtraArgs.D.getFunctionTypeInfo(); !NewFDisConst)
```

Comment:
```suggestion
      if (const auto &FTI = ExtraArgs.D.getFunctionTypeInfo(); NewFDisConst)
```
This is closer to what we mean, right?

Reply:
Er, no. We want to suggest removing `const` when `NewFDisConst` is `true`, or suggest inserting it if `false`.

Reply:
I think it would work either way... since this is guarded by `if (FDisConst != NewFDisConst)`, `!NewFDisConst` implies `FDisConst`.

Reply:
Yeah, I agree it is not really a meaningful change, besides being a bit of a 'say what you mean'.  That is, use the variable that is more closely related to what we MEAN.  I haven't reviewed this in long enough to know if I was correct here, so if you're sure that the variable you're using is the one that properly states the intent, feel free to ignore hte comment.